types/schemas/discriminated/ApplicationData.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  ApplicationTypeKeys,
+  PrimaryApplicationType,
+} from '../digitalPlanningApplication/enums/ApplicationTypes';
+
+/**
+ * @internal
+ * Base type for ApplicationData. Contains all shared properties across all application types
+ */
+export interface ApplicationDataBase<TGranular extends ApplicationTypeKeys> {
+  applicationType: TGranular;
```

Comment:
`TGranular` is repeated here in order to make this change slightly less disruptive. Note that this is just the string value and not the value/description object (we could still support this!).